src/methods/geom_relations/geom_geom_processors.jl

```diff
@@ -669,4 +707,27 @@ end
 function _point_in_extent(p, extent::Extents.Extent)
     (x1, x2), (y1, y2) = extent.X, extent.Y
     return x1 ≤ GI.x(p) ≤ x2 && y1 ≤ GI.y(p) ≤ y2
-end
\ No newline at end of file
+end
+
+# Disjoint extent optimisation: skip work based on geom extent intersection
+# returns Tuple{Bool, Bool} for (skip, returnval)
+@inline function _maybe_skip_disjoint_extents(a, b;
+    in_allow, on_allow, out_allow, 
+    in_require, on_require, out_require,
+    kw...
+)
```

Comment:
I don't think this logic is quite clear and I think it misses the cases where `in_allow = true` and `out_allow = true` but no `in` or `on` points are required. In that case, a `disjoint` could still short circuit and return a `true` and we could be done. I would suggest something like:

```julia
ext_disjoint = Extents.disjoint(GI.extent(a), GI.extent(b))
skip, return_val = if !ext_disjoint
    # can't tell anything about this case
    false, false
elseif out_allow  # && ext_disjoint
    if in_require || on_require
        true, false
    else
        true, true
    end
else  # !out_allow && ext_disjoint
    # points not allowed in exterior, but geoms are disjoint
    true, false
end
return skip, return_val
```

Something to note here is that you shouldn't be able to have a `required` val be `true` without the `allowed` val being `true`. I think that is implicitly checked in my code, but might need to be more explicit. But logically, you can't require a point in a given region if not points are allowed in that region.

Reply:
Ah yes I didn't quite get that required was a subset of allowed. And good to add more cases we can reject, thanks for the edit

Reply:
(And yes `GI.extent` checks for an extent first and calculates if it doesn't exist. `Extents.extent` cant do the calculating part as it doesn't know about geointerface objects, so we added `GI.extent` partly for this role)